packages/replay-next/components/elements-new/utils/serialization.ts

```diff
@@ -117,6 +117,8 @@ export function serializeDOM(rootNode: Document): number[] {
         const id = __RECORD_REPLAY_ARGUMENTS__.internal.registerPlainObject(value);
         if (id) {
           return parseInt(id);
+        } else {
+          throw Error(`[RUN-2965] registerPlainObject failed for value: ${JSON.stringify(value, null, 2)}`);
```

Comment:
I don't think there's anything that will pick up and report this error. In the case where the evaluation fails, the frontend degrades gracefully and shows "_Elements are not available_" in the Elements panel.

Reply:
Maybe the backend will automatically log thrown errors? (If so I didn't know that.)

The frontend can't. If an evaluation throws, the response that gets sent to the frontend is:
```js
{
  exception: {
    unavailable: true,
  },
  returned: {
    object: "1",
  },
}
```

I guess the frontend could add some scaffolding of its own to do this but I think that would be clunky.

Reply:
I think you are right, that it should be easier to have RUN triage their own problems.
I guess in this case, considering `registerPlainObject` should be infallable, so making it produce a RUN warning/error if it don't work, would make sense 🤔

Reply:
> considering registerPlainObject should be infallable, so making it produce a RUN warning/error if it don't work, would make sense

I _think_ you're saying that RUN would do this? (So it would generate its own logging from those failures?)

Sorry I'm not sure I understand. Maybe easier if we chat about this on VC briefly.